src/CodeStyle/Core/CodeFixes/AbstractFormattingAnalyzerImpl.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.CodingConventions;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    internal abstract class AbstractFormattingAnalyzerImpl
+    {
+        private readonly DiagnosticDescriptor _descriptor;
+
+        protected AbstractFormattingAnalyzerImpl(DiagnosticDescriptor descriptor)
+        {
+            _descriptor = descriptor;
+        }
+
+        internal void InitializeWorker(AnalysisContext context)
+        {
+            var workspace = new AdhocWorkspace();
+            var codingConventionsManager = CodingConventionsManagerFactory.CreateCodingConventionsManager();
+
+            context.RegisterSyntaxTreeAction(c => AnalyzeSyntaxTree(c, workspace, codingConventionsManager));
```

Comment:
>RegisterSyntaxTreeAction [](start = 20, length = 24)

Just an FYI that the analyzer driver prioritizes syntax tree action execution over all other actions, with the assumption that computing syntax diagnostics is generally the cheapest and also most critical to see early for end users. If this analyzer is slow, it will have a truly visible impact on perf.